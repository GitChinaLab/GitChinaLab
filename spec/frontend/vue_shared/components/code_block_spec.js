import { shallowMount } from '@vue/test-utils';
import CodeBlock from '~/vue_shared/components/code_block.vue';

describe('Code Block', () => {
  let wrapper;

  const defaultProps = {
    code: 'test-code',
  };

  const createComponent = (props = {}) => {
    wrapper = shallowMount(CodeBlock, {
      propsData: {
        ...defaultProps,
        ...props,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('with default props', () => {
    beforeEach(() => {
      createComponent();
    });

    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('with maxHeight set to "200px"', () => {
    beforeEach(() => {
      createComponent({ maxHeight: '200px' });
    });

    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
