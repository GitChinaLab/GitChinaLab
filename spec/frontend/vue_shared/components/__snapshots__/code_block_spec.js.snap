// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Block with default props renders correctly 1`] = `
<pre
  class="code-block rounded code"
>
  <code
    class="d-block"
  >
    test-code
  </code>
</pre>
`;

exports[`Code Block with maxHeight set to "200px" renders correctly 1`] = `
<pre
  class="code-block rounded code"
  style="max-height: 200px; overflow-y: auto;"
>
  <code
    class="d-block"
  >
    test-code
  </code>
</pre>
`;
